src/main/java/cashleh/parser/Parser.java

```diff
@@ -184,6 +205,80 @@ private Command getDeleteTransaction(String input, String transactionType) throw
             new DeleteExpense(transactionIndex, expenseStatement) : new DeleteIncome(transactionIndex, incomeStatement);
     }
 
+    /**
+     * Parses and extracts filtering criterion based on the specified transaction type and input string.
+     * @param transactionType The type of transaction to filter (FILTER_EXPENSE, FILTER_INCOME, or FILTER).
+     * @param input The input string to parse and filter.
+     * @return A {@link FindParser} containing the parsed and filtered transaction details.
+     * @throws CashLehParsingException If the input or parsing fails.
+     */
+    private FindParser filterDetails (String transactionType, String input) throws CashLehParsingException {
+        String[] format = null;
+        switch (transactionType) {
+        case FILTER_EXPENSE:
+            format = new String[]{FILTER_EXPENSE, "/amt:optional", "/date:optional", "/cat:optional"};
+            break;
+        case FILTER_INCOME:
+            format = new String[]{FILTER_INCOME, "/amt:optional","/date:optional", "/cat:optional"};
+            break;
+        case FILTER:
+            format = new String[]{FILTER, "/amt:optional", "/date:optional", "/cat:optional"};
+            break;
+        default:
+            throw new CashLehParsingException("Aiyoh! Your input blur like sotong... Clean your input for CashLeh!");
+        }
+        HashMap<String, String> inputDetails = StringTokenizer.tokenize(input, format);
+        String descriptionString = inputDetails.get(transactionType);
+        String amountString = inputDetails.get("/amt");
+        String dateString = inputDetails.get("/date");
+        String categoryString = inputDetails.get("/cat");
+
+        if ((descriptionString == null || descriptionString.isEmpty()) &&
+                (amountString == null || amountString.isEmpty()) &&
+                (dateString == null || dateString.isEmpty()) &&
+                (categoryString == null || categoryString.isEmpty())){
+            throw new CashLehParsingException("Please provide at least one filter criterion " +
+                    "(description, amount, date, or category)!");
+        }
+
+        double parsedAmount = -1.0;
```

Comment:
i think we can use a optional type and initialise it as null instead of using -1 as a magic number